### DISCUSSION TITLE
How to set stereo downmixing in SMPlayer's mplayer options?

### DISCUSSION THREAD

**Discussion by tonyrulez** [smplayer-dev/smplayer#5] [Q&A] [answered]
Downmixing 5.1 to stereo is not so good even with default settings as on PotPlayer for Windows. I found on an old forum a pan filter, which should be better:

`LC_ALL=C mplayer -af pan=6:.32:.39:.06:.17:-.17:.33:.32:.06:.39:-.17:.17:.33 a_file.mkv`

When putting only `-af pan=6:.32:.39:.06:.17:-.17:.33:.32:.06:.39:-.17:.17:.33` in the optons results in no sound. But `LC_ALL=C` should fix this issue. However when I put it in the mplayer options, then the video simply won't start.

How can I set this pan filter in the settings of SMPlayer?

**Comment by smplayer-dev** [comment, marked as answer]
In Preferences -> Advanced -> MPlayer/mpv, type this in the "audio filters" field:
`pan=6:.32:.39:.06:.17:-.17:.33:.32:.06:.39:-.17:.17:.33`

You also have to be using mplayer as multimedia engine (Preferences -> General).

**Reply by tonyrulez** [reply]
Thank you, it works this way. It was my bad, I didn't set it to mplayer, also `-af` was not needed.

**Comment by M4st0d0n** [comment]
Hello,

The pan filter was not working for me. There's an example in the end of mplayer's man where it's used for downmixing a 6 channel AAC to stereo, using pan=2, not pan=6 :

       Play a 6-channel AAC file with only two speakers:
       mplayer -rawaudio format=0xff -demuxer rawaudio -af pan=2:.32:.32:.39:.06:.06:.39:.17:-.17:-.17:.17:.33:.33 adts_he-aac160_51.aac
       You might want to play a bit with the pan values (e.g multiply  with  a
       value) to increase volume or avoid clipping.

For my video I ended up using to force decoding the 6 channel and then downmix them with the pan filter. I'm still confused about how it works, I guess it's the number of channels in the first parameter then a routing matrix for the six channels with negative value feeding phase switched signal. With channel order being Center, Front L, Front R, Back L, back R and LFE (not sure about that). I also added a filter to boost by 10dB because the sound was really low, but I monitored the level on easyeffect and it seemed there were no clipping.

mplayer -channels 6 -af pan=2:.32:.32:.39:.06:.06:.39:.17:-.17:-.17:.17:.33:.33,volume=10

**Comment by M4st0d0n** [comment]
Okay, I tested that further with pan=2:1:0:0:0:0:0:0:0:0:0:0:0, then pan=2:0:1:0:0:0:0:0:0:0:0:0:0, then pan=2:0:0:1:0:0:0:0:0:0:0:0:0, etc.

So the level routing matrix parameters are definately grouped by channels, but for my video the orders are not the same that were described in mplayer man example (mine is an ac3 surround of an mkv file). With the tweaks I think it's Front L, Front R, Back L, Back R, Center and LFE.

I've modified the filter parameters

pan=2:.39:.06:.06:.39:.17:-.17:-.17:.17:.32:.32:.33:.33,volume=10

I think the results are OK, voices are a lot better now and stereo image in actions scenes seems to be alright.

**Comment by urapoh** [comment]
I noticed that in my instance the filter doesn't work as it should, i.e. only the first 5 parameters (FR and FL source channels) are working. 
My solution is to change the multimedia engine to mpv. With mpv pan filter works fine, though it has different syntax. For example that is my simple filter config for enhancing dialogs and reducing SFX:
`pan=stereo| FL < FL + FC - BL - SL + 0.5*LFE | FR < FR + FC - BR - SR + 0.5*LFE`
